x-pack/legacy/plugins/epm/server/lib/kibana/index_pattern/install.test.ts

```diff
@@ -21,18 +21,115 @@ expect.addSnapshotSerializer({
   },
 });
 const files = glob.sync(path.join(__dirname, '/tests/*.yml'));
-let allFields: Field[] = [];
+let fields: Fields = [];
 for (const file of files) {
   const fieldsYML = readFileSync(file, 'utf-8');
-  allFields = allFields.concat(safeLoad(fieldsYML));
+  fields = fields.concat(safeLoad(fieldsYML));
 }
 
 test('flattenFields function recursively flattens nested fields and renames name property with path', () => {
-  const flattened = flattenFields(allFields);
+  const flattened = flattenFields(fields);
   expect(flattened).toMatchSnapshot('flattenFields');
 });
 
 test('dedupFields function remove duplicated fields when parsing multiple files', () => {
-  const deduped = dedupFields(allFields);
+  const deduped = dedupFields(fields);
   expect(deduped).toMatchSnapshot('dedupFields');
 });
+
+test('transformField maps field types to kibana index pattern data types', () => {
+  expect(transformField({ name: 'testField' }).type).toBe('string');
+  expect(transformField({ name: 'testField', type: 'half_float' }).type).toBe('number');
+  expect(transformField({ name: 'testField', type: 'scaled_float' }).type).toBe('number');
+  expect(transformField({ name: 'testField', type: 'float' }).type).toBe('number');
+  expect(transformField({ name: 'testField', type: 'integer' }).type).toBe('number');
+  expect(transformField({ name: 'testField', type: 'long' }).type).toBe('number');
+  expect(transformField({ name: 'testField', type: 'short' }).type).toBe('number');
+  expect(transformField({ name: 'testField', type: 'byte' }).type).toBe('number');
+  expect(transformField({ name: 'testField', type: 'keyword' }).type).toBe('string');
+  expect(transformField({ name: 'testField', type: 'string' }).type).toBe(undefined);
+  expect(transformField({ name: 'testField', type: 'text' }).type).toBe('string');
+  expect(transformField({ name: 'testField', type: 'date' }).type).toBe('date');
+  expect(transformField({ name: 'testField', type: 'geo_point' }).type).toBe('geo_point');
+  expect(transformField({ name: 'testField', type: 'invalid' }).type).toBe(undefined);
+});
+
+test('transformField changes values based on other values', () => {
```

Comment:
The stanzas/paragraphs feel like suites instead of tests because there are so many `expect`s. How do you feel about adding `describe('searchable', ...)`, `describe('aggregatable', ...)`, etc?

Reply:
I see it more as one test with a lot of expects.  If it was describe, then there would still be one test in each describe, unless i wrapped each expect in a test with a description which might be overkill.

Reply:
I meant to write that the `test`s seemed like a suite and the stanzas/paragraphs seemed more like `test`s.

It would still have lots of `expect`s per `test`. It would just go from 40+ for a single `test` to +/- 5 in multiple `test`s.

e.g.

```
describe('transformField changes values based on other values', () => {

  test('searchable', () => {
    expect(transformField({ name: 'testField' }).searchable).toBe(true);
    expect(transformField({ name: 'testField', searchable: true }).searchable).toBe(true);
    expect(transformField({ name: 'testField', searchable: false }).searchable).toBe(false);
    expect(transformField({ name: 'testField', type: 'binary' }).searchable).toBe(false);
    expect(transformField({ name: 'testField', searchable: true, type: 'binary' }).searchable).toBe(
      false
    );
  });

  // ...

  test('indexed', () => {
    expect(transformField({ name: 'testField', type: 'binary' }).indexed).toBe(false);
    expect(transformField({ name: 'testField', index: true, type: 'binary' }).indexed).toBe(false);
  });

  // etc

});
```

I get that one `expect` per `test` would be brutal, but I don't think we need to drop all grouping/description. A test might not fail for "only one reason" as some like, but at least it would be limited to a smaller facet of the code.